test/vault_helpers.go

```diff
@@ -428,57 +282,56 @@ func runVaultEC2AuthTest(t *testing.T, packerBuildName string) {
 	exampleSecret := "42"
 
 	defer test_structure.RunTestStage(t, "teardown", func() {
-		terraformOptions := test_structure.LoadTerraformOptions(t, examplesDir)
-		terraform.Destroy(t, terraformOptions)
+		teardownResources(t, examplesDir)
+	})
 
-		amiId := test_structure.LoadAmiId(t, examplesDir)
-		awsRegion := test_structure.LoadString(t, examplesDir, SAVED_AWS_REGION)
-		aws.DeleteAmi(t, awsRegion, amiId)
+	test_structure.RunTestStage(t, "setup_ami", func() {
+		setupAmi(t, examplesDir, packerBuildName)
+	})
 
-		keyPair := test_structure.LoadEc2KeyPair(t, examplesDir)
-		aws.DeleteEC2KeyPair(t, keyPair)
+	test_structure.RunTestStage(t, "deploy", func() {
+		uniqueId := random.UniqueId()
+		terraformVars := map[string]interface{}{
+			VAR_VAULT_AUTH_SERVER_NAME: fmt.Sprintf("vault-auth-test-%s", uniqueId),
+			VAR_VAULT_SECRET_NAME:      exampleSecret,
+		}
+		deployCluster(t, examplesDir, uniqueId, terraformVars)
+	})
 
-		tlsCert := loadTlsCert(t, examplesDir)
-		cleanupTlsCertFiles(tlsCert)
+	test_structure.RunTestStage(t, "validate", func() {
+		terraformOptions := test_structure.LoadTerraformOptions(t, examplesDir)
+		testRequestSecret(t, terraformOptions, exampleSecret)
 	})
+}
 
-	test_structure.RunTestStage(t, "setup_ami", func() {
-		awsRegion := aws.GetRandomRegion(t, nil, nil)
-		test_structure.SaveString(t, examplesDir, SAVED_AWS_REGION, awsRegion)
+// Test the Vault IAM authentication example by:
+//
+// 1. Copying the code in this repo to a temp folder so tests on the Terraform code can run in parallel without the
+//    state files overwriting each other.
+// 2. Building the AMI in the vault-consul-ami example with the given build name
+// 3. Deploying that AMI using the example Terraform code setting an example secret
+// 4. Waiting for Vault to boot, unseal and the client to fetch the secret
+// 5. Making a request to the webserver started by the auth client user data script to see if the secret was properly fetched
```

Comment:
Nice description of the test, but it doesn't mention Vault IAM auth explicitly anywhere? Perhaps clarify in the last line that you're using Vault IAM Auth to auth the client? Also can you make it clear that you're not "waiting for vault" to unseal; rather, the test will unseal Vault.

Reply:
ok!